packages/react-reconciler/src/ReactFiberCommitWork.new.js

```diff
@@ -1868,6 +1871,212 @@ function commitPassiveLifeCycles(finishedWork: Fiber): void {
   }
 }
 
+function commitDoubleInvokeEffectsInDEV(
+  fiber: Fiber,
+  hasPassiveEffects: boolean,
+) {
+  if (enableDoubleInvokingEffects) {
+    if (__DEV__) {
+      invokeLayoutEffectsUnmountInDEV(fiber);
+      if (hasPassiveEffects) {
+        invokePassiveEffectsUnmountInDEV(fiber);
+      }
+
+      invokeLayoutEffectsMountInDEV(fiber);
+      if (hasPassiveEffects) {
+        invokePassiveEffectsMountInDEV(fiber);
+      }
+    }
+  }
+}
+
+function invokeLayoutEffectsUnmountInDEV(firstChild) {
+  if (enableDoubleInvokingEffects) {
+    if (__DEV__) {
+      // unmount layout effects
+      let fiber = firstChild;
+      while (fiber !== null) {
+        if (fiber.child !== null) {
+          // Should we add a separate subtree tag for this?
```

Comment:
What's this comment for? I'm not sure I understand it.

Is this pointing out the fact that we're going to end up traversing all of the subtree paths that have layout effects *even if they're all updates and none of them are mounts*? So we could add a DEV-only tag e.g. `MountLayoutSubtreeTag` or something.

Maybe we could add some additional clarification to the wording here.